core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -226,7 +227,14 @@ public String getIcon() {
      */
     @Deprecated @Override
     public boolean isJnlpAgent() {
-        return launcher instanceof JNLPLauncher;
+        final Class<?> jnlpLauncherClass;
+        try {
+            jnlpLauncherClass = Class.forName("hudson.slaves.JNLPLauncher");
+        } catch (Throwable ex) {
+            // Ignore
+            return false;
+        }
+        return jnlpLauncherClass.isAssignableFrom(launcher.getClass());
```

Comment:
I know, but there are no expected subclasses. Anyway this is good enough if verbose.